pkg/crypto/crypto_property_test.go

```diff
@@ -0,0 +1,270 @@
+package crypto
+
+import (
+	"crypto/ed25519"
+	crypto_rand "crypto/rand"
+	"math/rand"
+	"reflect"
+	"testing"
+	"testing/quick"
+)
+
+type keyPair struct {
+	PublicKey  ed25519.PublicKey
+	PrivateKey ed25519.PrivateKey
+}
+
+// Generate implements quick.Generator for keyPair
+func (keyPair) Generate(rand *rand.Rand, size int) reflect.Value {
+	pub, priv, err := ed25519.GenerateKey(crypto_rand.Reader)
+	if err != nil {
+		panic(err)
+	}
+	kp := keyPair{
+		PublicKey:  pub,
+		PrivateKey: priv,
+	}
+	return reflect.ValueOf(kp)
+}
+
+// randomMessage generates a random byte slice for testing
+type randomMessage []byte
+
+// Generate implements quick.Generator for randomMessage
+func (randomMessage) Generate(rand *rand.Rand, size int) reflect.Value {
+	// Generate message size between 1 and 1024 bytes
+	msgSize := 1 + rand.Intn(1024)
+	message := make([]byte, msgSize)
+	rand.Read(message)
+	return reflect.ValueOf(randomMessage(message))
+}
+
+// Property-based tests using testing/quick
+
+func TestEncryptionDecryptionRoundtrip(t *testing.T) {
+	f := func(kp keyPair, msg randomMessage) bool {
+		message := []byte(msg)
+		encrypted, err := Encrypt(message, kp.PublicKey)
```

Comment:
extract that logic into a function e.g encryptThenDecrypt and the only thing would be left is comparing the decrypted with the message